src/java.base/unix/native/libjli/java_md.c

```diff
@@ -643,11 +644,24 @@ void* SplashProcAddress(const char* name) {
     }
 }
 
+// Instead of waiting on the implicit futex that is located in the address
+// space of the main Java thread stack we'll wait on a global condition here.
+static pthread_mutex_t main_thread_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t main_thread_cond = PTHREAD_COND_INITIALIZER;
+static bool main_thread_done = false;
+static int main_thread_result = 0;
+
 /*
  * Signature adapter for pthread_create() or thr_create().
  */
 static void* ThreadJavaMain(void* args) {
-    return (void*)(intptr_t)JavaMain(args);
+    main_thread_result = JavaMain(args);
```

Comment:
I think this will work, but let's assign and read `main_thread_result` in the critical section. To avoid the need for reasoning why main_thread_result is visible in other thread.